### ISSUE TITLE
Presentation to command translators sometimes don't execute the command

### ISSUE THREAD

**Issue by scymtym** [McCLIM/McCLIM#1080]
When an interactor pane is present (the problem may also occur without that, but the interactor pane makes it easier to observe), clicking a presentation for which a presentation to command translator is applicable (and has the highest priority, etc.), the command is sometimes echoed in the interactor and executed. In some cases however, the command is echoed in the interactor (or inserted for input editing to be more precise) but not executed.

An easy way to reproduce the issue is the tab layout demo. Clicking a tab header only sometimes switches to the tab as described above.

Based on bisection, the bad commit seems to be ec2225c06 ("accept command: provide input context to click on a command").

**Comment by dkochmanski** [comment]
I'm trying to reproduce this, but clicking on a tab header in the "Tab  Layout" demo (also after randomization and adding new panes), works correctly so far.

**Comment by scymtym** [comment]
It is a little hard to see, but the final click onto the header of the "B" tab at the end of the video inserts the command into the interactor without executing it:
![success-master](https://user-images.githubusercontent.com/150189/95170514-dab90b80-07b4-11eb-9eed-cce26cc955ed.gif)

**Comment by dkochmanski** [comment]
READ-COMMAND calls (indirectly) ACCEPT-1 with the type COMMAND.

ACCEPT-1 function (pseudocode):

#+BEGIN_SRC lisp
  (defun accept-1 (...)
    (with-input-context ('command)
        (object ptype)
        (multiple-value-bind (result result-type)
            (funcall-presentation-generic-function accept ptype)
          (unless *recursive-accept-p*
            (when-let ((gesture (read-gesture :stream stream :timeout 0)))
              (unless (activation-gesture-p gesture)
                (unread-gesture gesture :stream stream))))
          (values result result-type))
      (t
       (values object ptype))))
#+END_SRC

The accept method for the type command old version (pseudocode):
#+BEGIN_SRC lisp
  (define-method accept (...)
    (values (command-parser) 'command))
#+END_SRC

The accept method for the type command new version (pseudocode)
#+BEGIN_SRC lisp
  (define-method accept (...)
    (with-input-context ('command)
       (object ptype)
       (values (command-parser) 'command)
      (command
       (values object ptype))))
#+END_SRC

With the old version, when the command was clicked, then the control was
transferred from the body of the with-input-context to the clause T, so
read-gesture was never executed. Currently, when the command is clicked,
funcall-presentation-generic-function returns and we advance to the call of
read-gesture. Mind, that since we have clicked on the presentation, there is
no "activation gesture" in it.

When the interactor is present, then read-gesture in accept-1 is called on the
input-editing-stream which encapsulates the interactor. That leads to a call
of a STREAM-READ-GESTURE method specialized on DREI-INPUT-EDITING-STREAM
(defined in Libraries/Drei/input-editor.lisp). Pseudocode follows:

#+BEGIN_SRC lisp
  (defmethod stream-read-gesture ((stream drei-input-editing-mixin) &rest args)
    (loop (do-input-editor-processing)
          (loop (multiple-value-bind (gesture type)
                    (stream-read-command encapsulated-stream :timeout 0)
                  (when (null gesture)
                    (return-from stream-read-gesture (values gesture type)))
                  (when (stream-process-gesture stream gesture type)
                    (loop-finish))))))

  (defmethod stream-process-gesture ((stream drei-input-editing-mixin) gesture type)
    ;; this function calls to esa-read-gesture, which in turn calls read-gesture
    ;; in loop until a "proper gesture" arrives, that is the input editing gesture.
    (read-gestures-and-act))
#+END_SRC

Recently we've refactored input streams so the input buffer is not the same
object as the event queue (for conformance and glory), stream-read-gesture
called on the interactor pane may return i.e a pointer event.

The failure scenario is as follows:
1. user clicks on a command
2. read-gesture in accept-1 is called
   >   (stream-read-gesture drei-stream :timeout 0)
   >>  (stream-read-gesture interactor  :timeout 0)
   <<  <pointer-event>
   >>  (stream-process-gesture :no-timeout-whatsoever t)
   >>> (esa-wait-for-keyboard-gesture)
   >>> ...

The success scenario is as follows
1. user clicks on a command
2. read-gesture in accept-1 is called
   >   (stream-read-gesture drei-stream :timeout 0)
   >>  (stream-read-gesture interactor  :timeout 0)
   <<  (values nil :timeout)
   <   (values command 'command)

In other words, if a pointer event is available in the input buffer after the command click, esa waits for the next keyboard press.
